qt/scientific_interfaces/Inelastic/Manipulation/InelasticDataManipulationSymmetriseTabView.cpp

```diff
@@ -175,103 +169,101 @@ IndirectPlotOptionsView *InelasticDataManipulationSymmetriseTabView::getPlotOpti
-    if (eMax < 0) {
-      eMax = -eMax;
-      m_dblManager->setValue(m_properties["EMax"], eMax);
-      return;
-    }
-    // If range is invalid reset EMax to double EMin
-    else if (eMin > eMax) {
-      m_dblManager->setValue(m_properties["EMax"], eMin * 2);
-      return;
-    }
-  }
-
-  // If we get this far then the E range is valid
-  // Update the range selectors with the new values.
-  updateRangeSelectors(prop, value);
+void InelasticDataManipulationSymmetriseTabView::xRangeLowChanged(double value) {
+  m_dblManager->setValue(m_properties["Elow"], value);
+  m_dblManager->setMinimum(m_properties["Ehigh"], value);
 }
 
 /**
- * Updates position of XCut range selectors when used changed value of XCut.
+ * Handles the X maximum value being changed from a range selector.
  *
- * @param prop QtProperty changed
- * @param value Value it was changed to (unused)
+ * @param value New range selector value
  */
-void InelasticDataManipulationSymmetriseTabView::updateRangeSelectors(QtProperty *prop, double value) {
-  auto positiveERaw = m_uiForm.ppRawPlot->getRangeSelector("PositiveE");
-
-  value = fabs(value);
-
-  if (prop == m_properties["EMin"]) {
-    positiveERaw->setMinimum(value);
-  } else if (prop == m_properties["EMax"]) {
-    positiveERaw->setMaximum(value);
-  }
+void InelasticDataManipulationSymmetriseTabView::xRangeHighChanged(double value) {
+  m_dblManager->setValue(m_properties["Ehigh"], value);
+  m_dblManager->setMaximum(m_properties["Elow"], value);
 }
 
 /**
- * Handles the X minimum value being changed from a range selector.
+ * Updates boundaries and initial values for Selector and Data properties when changing between negative/positive side
+ * of spectrum.
+ *
+ * @param isPositive flag for spectrum positive side
+ * @param range Active spectra range
  *
- * @param value New range selector value
  */
-void InelasticDataManipulationSymmetriseTabView::xRangeMinChanged(double value) {
-  m_dblManager->setValue(m_properties["EMin"], std::abs(value));
-  m_uiForm.pbPreview->setEnabled(true);
+void InelasticDataManipulationSymmetriseTabView::resetEDefaults(bool isPositive, QPair<double, double> range) {
+  auto rangeESelector = m_uiForm.ppRawPlot->getRangeSelector("rangeE");
+  auto rangeReflect = isPositive ? QPair(0.0, range.second) : QPair(range.first, 0.0);
+  auto rangeInitial =
+      isPositive ? QPair(0.1 * range.second, 0.9 * range.second) : QPair(0.9 * range.first, 0.1 * range.first);
+
+  rangeESelector->setBounds(rangeReflect.first, rangeReflect.second);
+  m_dblManager->setRange(m_properties["Ehigh"], rangeReflect.first, rangeReflect.second);
+  m_dblManager->setRange(m_properties["Elow"], rangeReflect.first, rangeReflect.second);
+
+  rangeESelector->setRange(rangeInitial.first, rangeInitial.second);
+  m_dblManager->setValue(m_properties["Ehigh"], rangeInitial.second);
+  m_dblManager->setValue(m_properties["Elow"], rangeInitial.first);
```

Comment:
This part is a little bit confusing, I suggest refactoring like this to make it more readable. I guess variable naming could be better
```suggestion
  auto rangeESelector = m_uiForm.ppRawPlot->getRangeSelector("rangeE");
  
  // set selector range boundaries
  auto rangeReflect = isPositive ? QPair(0.0, range.second) : QPair(range.first, 0.0);

  rangeESelector->setBounds(rangeReflect.first, rangeReflect.second);
  m_dblManager->setRange(m_properties["Ehigh"], rangeReflect.first, rangeReflect.second);
  m_dblManager->setRange(m_properties["Elow"], rangeReflect.first, rangeReflect.second);

   // set initial selector range values
   auto rangeInitial =
      isPositive ? QPair(0.1 * range.second, 0.9 * range.second) : QPair(0.9 * range.first, 0.1 * range.first);
      
  rangeESelector->setRange(rangeInitial.first, rangeInitial.second);
  m_dblManager->setValue(m_properties["Ehigh"], rangeInitial.second);
  m_dblManager->setValue(m_properties["Elow"], rangeInitial.first);
```

Reply:
thanks for the suggestion